src/GafferImage/ImageWriter.cpp

```diff
@@ -706,10 +711,24 @@ ImageWriter::ImageWriter( const std::string &name )
 	addChild( new ImagePlug( "in" ) );
 	addChild( new StringPlug( "fileName" ) );
 	addChild( new StringPlug( "channels", Gaffer::Plug::In, "*" ) );
+
+	ColorSpacePtr colorSpace = new ColorSpace( "__colorSpace" );
+	addChild( colorSpace );
+	OpenColorIO::ConstConfigRcPtr config = OpenColorIO::GetCurrentConfig();
+	colorSpace->inputSpacePlug()->setValue( config->getColorSpace( OpenColorIO::ROLE_SCENE_LINEAR )->getName() );
+	colorSpace->inPlug()->setInput( inPlug() );
+
+	ExpressionPtr expression = new Expression( "__expression" );
+	addChild( expression );
+	//expression->setExpression( string( "parent[\"__colorSpace\"][\"outputSpace\"] = context.get(\"gaffer:colorspace\", \"\") " ), string( "python" ) );
+
+	addChild( new StringPlug( "colorSpace" ) );
```

Comment:
This is a minor thing, but generally I've found it more readable to add all the public plugs _first_, at the top of the function, and then all the implementation details after (in which case this would move to line 714). That gets you into a "here's my interface, and here's my implementation" mindset, and also means the order of the plug/node accessors in the .cpp file matches that in the header.